pkg/sources/gitlab/gitlab.go

```diff
@@ -203,6 +210,68 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk, _ .
 	return s.scanRepos(ctx, chunksChan)
 }
 
+func (s *Source) scanTargets(ctx context.Context, client *gitlab.Client, targets []sources.ChunkingTarget, chunksChan chan *sources.Chunk) error {
+	for _, tgt := range targets {
+		if err := s.scanTarget(ctx, client, tgt, chunksChan); err != nil {
+			ctx.Logger().Error(err, "error scanning target")
+		}
+	}
+
+	return nil
+}
+
+func (s *Source) scanTarget(ctx context.Context, client *gitlab.Client, target sources.ChunkingTarget, chunksChan chan *sources.Chunk) error {
+	metaType, ok := target.QueryCriteria.GetData().(*source_metadatapb.MetaData_Gitlab)
+	if !ok {
+		return fmt.Errorf("unable to cast metadata type for targeted scan")
+	}
+	meta := metaType.Gitlab
+	projID, sha := int(meta.GetProjectId()), meta.GetCommit()
+	if projID == 0 || sha == "" {
+		return fmt.Errorf("project ID and commit SHA must be provided for targeted scan")
+	}
+
+	aCtx := context.WithValues(ctx, "project_id", projID, "commit", sha)
+
+	diffs, err := s.getCommitDiffs(aCtx, client, projID, sha)
+	if err != nil {
+		return fmt.Errorf("error fetching diffs for commit %s: %w", sha, err)
+	}
+
+	for _, diff := range diffs {
+		if diff.Diff == "" {
+			aCtx.Logger().V(4).Info("skipping empty diff", "file", diff.NewPath)
+			continue
+		}
+
+		chunk := &sources.Chunk{
+			SourceType: s.Type(),
+			SourceName: s.name,
+			SourceID:   s.SourceID(),
+			JobID:      s.JobID(),
+			SecretID:   target.SecretID,
+			Data:       []byte(diff.Diff),
+			SourceMetadata: &source_metadatapb.MetaData{
+				Data: &source_metadatapb.MetaData_Gitlab{Gitlab: meta},
+			},
+			Verify: s.verify,
+		}
+
+		if err := common.CancellableWrite(ctx, chunksChan, chunk); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *Source) getCommitDiffs(ctx context.Context, client *gitlab.Client, projID int, sha string) ([]*gitlab.Diff, error) {
+	const paginationLimit = 100 // Default is 20, max is 100.
+	diffOpts := &gitlab.GetCommitDiffOptions{ListOptions: gitlab.ListOptions{PerPage: paginationLimit}}
+	diffs, _, err := client.Commits.GetCommitDiff(projID, sha, diffOpts, gitlab.WithContext(ctx))
```

Comment:
Isn't this query paginated? Do we need to look beyond the first page?

Reply:
My understanding from reading the docs suggest it is not paginated. https://docs.gitlab.com/ee/api/commits.html#get-the-diff-of-a-commit
There doesn't appear to be a way to get another page, or cursor, so I think we should be okay with this approach.
Usually the request would include the `page_token` attribute if pagination was allowed.

Reply:
probably just copied it from above. will remove